packages/@aws-cdk/aws-secretsmanager/lib/secret.ts

```diff
@@ -707,6 +711,39 @@ function parseSecretName(construct: IConstruct, secretArn: string) {
   throw new Error('invalid ARN format; no secret name provided');
 }
 
+/**
+ * Parses the secret name from the ARN of an owned secret. With owned secrets we know that a few things we don't with imported secrets:
+ * - The ARN is guaranteed to be a full ARN, with suffix.
+ * - The name -- if provided -- will tell us how many hyphens to expect in the final secret name.
+ * - If the name is not provided, we know the format used by CloudFormation for auto-generated names.
+ *
+ * Note: This is done rather than just returning the secret name passed in by the user to keep the relationship
```

Comment:
Btw, I was just discussing with @iliapolo the idea of introducing a feature in core of tokens that encode a dependency. Similarly to how cross stack references work, The framework will identify these tokens, resolve them to their concrete value and automatically add a dependency between the two resources.

This way, we won't have to be limited to use CfN intrinsic to enforce the dependency